### PR TITLE
feat(perf): Track user updates to the search query across performance and discover

### PR DESCRIPTION
**Pull request by udameli**
This PRs tracks when users update a search query by appending `isDirty` query parameter to the url. This is done for search bars on the `Performance landing page`, `Transaction Summary`, `Transaction All Events`, `Spans`, `Spans Summary`, `Vital Detail`, `Discover Results` pages.

Below are the steps that user might take and how search bar state is tracked:
- user is on the performance landing page
- `isDirty: false`
- `/events-stats` and `/eventsv2` receive `user_modified: false`
- `query.user_modified` is set to `false` in the backend for those endpoints
- user types something in the search bar
- `isDirty: true`
- `/events-stats` and `/eventsv2` calls are made with `user_modified: true`
-  `query.user_modified` tag is set to 'true'
- user finds a transaction they want and go to the transaction summary page
- `isDirty: false`
- user types something in the search bar
- `isDirty: true`
- ... (same behavior for all other pages)

Fixes PERF-1563

### CODE REVIEW

File: static/app/views/performance/index.tsx [hunk 1]
```diff
@@ -8,10 +12,30 @@ import withOrganization from 'sentry/utils/withOrganization';
...
+  useEffect(
+    function () {
+      // when new perf page loads, query is pristine
+      if (location.pathname !== prevLocationPathname.current) {
+        prevLocationPathname.current = location.pathname;
+        browserHistory.push({
+          pathname: location.pathname,
+          query: {
+            ...location.query,
+            isDirty: false,
```
Comment (k-fish) [inline static/app/views/performance/index.tsx:31]:
```suggestion
            isDirty: undefined,
```

File: static/app/views/eventsV2/index.tsx [hunk 2]
```diff
@@ -7,10 +10,30 @@ import withOrganization from 'sentry/utils/withOrganization';
...
+  useEffect(
+    // when new discover page loads, query is pristine
+    function () {
+      if (location.pathname !== prevLocationPathname.current) {
+        prevLocationPathname.current = location.pathname;
+        browserHistory.push({
+          pathname: location.pathname,
+          query: {
+            ...location.query,
+            isDirty: false,
```
Comment (k-fish) [inline static/app/views/eventsV2/index.tsx:29]: lets just leave it as `undefined/ null`, it keeps the url cleaner that way. 
```suggestion
            isDirty: undefined,
```

File: static/app/views/eventsV2/index.tsx [hunk 3]
```diff
@@ -7,10 +10,30 @@ import withOrganization from 'sentry/utils/withOrganization';
...
+          },
+        });
+      }
+    },
+    [location]
```
Comment (k-fish) [inline static/app/views/eventsV2/index.tsx:34]: maybe double check these top level useEffects aren't causing any rerenders?

**Review by k-fish — APPROVED** [review]
Looks good, though perhaps change `isDirty` to something like `searchModified` or `userModified` since it's more clear what is changed and closer to the api param.

File: static/app/views/eventsV2/index.tsx (hunk 3 above)
Comment (udameli) [reply static/app/views/eventsV2/index.tsx:34]: it seeeeems okay, no unnecessary rerenders noticed

**Pull request merged**